packages/tests/src.ts/test-utils.spec.ts

```diff
@@ -629,103 +456,151 @@ describe("Test Signature Manipulation", function() {
-    const tests: Array<TestCase.TypedTransaction> = loadTests("typed-transactions");
-
-    function equalsData(name: string, a: any, b: any, ifNull?: any): boolean {
-        assert.equal(ethers.utils.hexlify(a), ethers.utils.hexlify((b == null) ? ifNull: b), name);
-        return true;
-    }
-
-    function equalsNumber(name: string, a: any, b: any, ifNull?: any): boolean {
-        assert.ok(ethers.BigNumber.from(a).eq((b == null) ? ifNull: b), name);
-        return true;
-    }
-
-    function equalsArray(name: string, a: any, b: any, equals: (name: string, a: any, b: any) => boolean): boolean {
-        assert.equal(a.length, b.length, `${ name }.length`);
-        for (let i = 0; i < a.length; i++) {
-            if (!equals(`${ name }[${ i }]`, a[i], b[i])) { return false; }
-        }
-        return true;
-    }
-
-    function makeEqualsArray(equals: (name: string, a: any, b: any) => boolean): (name: string, a: any, b: any) => boolean {
-        return function(name: string, a: any, b: any) {
-            return equalsArray(name, a, b, equals);
-        };
-    }
-
-    function equalsAccessList(name: string, a: Array<any>, b: Array<any>): boolean {
-        return equalsArray(`${ name }-address`, a.map((f) => f.address), b.map((f) => f.address), equalsData) &&
-               equalsArray(`${ name }-storageKeys`, a.map((f) => f.storageKeys), b.map((f) => f.storageKeys), makeEqualsArray(equalsData))
-    }
-
-    function allowNull(name: string, a: any, b: any, equals: (name: string, a: any, b: any) => boolean): boolean {
-        if (a == null) {
-            assert.ok(b == null, `${ name }:!NULL`);
-            return true;
-        } else if (b == null) {
-            assert.fail(`${ name }:!!NULL`);
+    const sendingAccount = "0.0.101010";
+    it('Should parse ContractCreate', async function() {
+
+        const cc = new ContractCreateTransaction()
+            .setContractMemo("memo")
+            .setGas(1000)
+            .setBytecodeFileId("0.0.111111")
+            .setNodeAccountIds([new AccountId(0,0,3)])
+            .setConstructorParameters(new ContractFunctionParameters().addUint256(100))
+            .setTransactionId(TransactionId.generate(sendingAccount))
+            .freeze();
+        const tx =await  ethers.utils.parseTransaction(cc.toBytes());
+        assert(tx.gasLimit.toNumber() === 1000, "Invalid gas limit");
+        assert(tx.data == hexlify(cc.constructorParameters));
+    });
+
+    it("Should parse ContractExecute", async function() {
+        const ce = new ContractExecuteTransaction()
+            .setGas(1000)
+            .setContractId("0.0.1112121")
+            .setFunction("mint", new ContractFunctionParameters().addUint256(200000))
+            .setTransactionId(TransactionId.generate(sendingAccount))
+            .setNodeAccountIds([new AccountId(0,0,3)])
+            .freeze();
+        const tx = await ethers.utils.parseTransaction(ce.toBytes());
```

Comment:
I think that we should assert that:
- `hash` is the same as derived from cc
- `from` is correctly populated
- `value` is populated correctly (we should set the amount in the ContractExecute transaction)